app/webpack/users/edit/components/relationships.jsx

```diff
@@ -0,0 +1,205 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Pagination from "rc-pagination";
+import moment from "moment";
+
+import SettingsItem from "./settings_item";
+import RelationshipsCheckboxContainer from "../containers/relationships_checkbox_container";
+import UserFollowing from "./user_following";
+import BlockedMutedUsersContainer from "../containers/blocked_muted_users_container";
+
+const Relationships = ( {
+  relationships,
+  filteredRelationships,
+  updateFilters,
+  sortRelationships,
+  showModal,
+  loadPage,
+  page
+} ) => {
+  const showRelationships = ( ) => filteredRelationships.slice(
+    page * 10 - 10, page * 10 // only display up to 10 items on current page
```

Comment:
Does this mean API pagination isn't working? Theoretically you should be able to make it 10 per page with `per_page=10` in your request.

Reply:
The API pagination is working, but I couldn't figure out a good approach to be able to use the API pagination and also apply filters and sorting.

If a user sorts by A-Z, for example, I'm assuming the user would want to see the first ten results with names starting with A, B, C, etc. across *all* of their relationships, but if I'm using `per_page=10` I would only have enough information to apply that sorting for the 10 names currently on display.

I'm also making an assumption that if a user is on page 3 and chooses an additional sort/filter, he/she would automatically start back at page 1 to see the newest sorted/filtered results... but maybe I need more clarity on how the UI is supposed to work here.

Reply:
IMO, if the user changes the search, sort, or other filters, they should go back to page 1, and all of those should be performed in the API request, not the client. So if the user wants to sort by name, you would request `/v1/relationships?order_by=name&order=asc&page=1`.

That said, another approach might be to load all the user's relationships and then perform these operations in the client. For most users that will be fine. For people with a lot of relationships, there will be some cost in bandwidth and data held in memory.